lib/services/nativescript-project-plugins-service.ts

```diff
@@ -0,0 +1,463 @@
+		}).future<boolean>()();
+	}
+
+	private installLocalPlugin(pluginPath: string): IFuture<void> {
+		return (() => {
+			let pathToPlugin = path.resolve(pluginPath);
+			let content = this.$fs.readJson(path.join(pathToPlugin, this.$projectConstants.PACKAGE_JSON_NAME)).wait();
+			let name = content.name;
+
+			if(pathToPlugin.indexOf(this.$project.getProjectDir().wait()) === -1) {
+				let pathToInstall = path.join(this.$project.getProjectDir().wait(), "plugins");
+				this.$logger.printMarkdown(`Moving \`${pathToPlugin}\` directory to \`${pathToInstall}\` in order to be able to use the plugin in your project.`);
+				shelljs.cp("-Rf", pathToPlugin, pathToInstall);
+				pathToPlugin = path.join(pathToInstall, path.basename(pathToPlugin));
+			}
+
+			let pathToPackageJson = this.getPathToProjectPackageJson().wait();
+			let packageJsonContent = this.getProjectPackageJsonContent().wait();
+			packageJsonContent.dependencies[name] = path.relative(this.$project.getProjectDir().wait(), pathToPlugin);
+			this.$fs.writeJson(pathToPackageJson, packageJsonContent).wait();
+			this.$logger.info("Successfully added plugin " + name);
+		}).future<void>()();
+	}
+
+	private findPluginsByName(name: string): IFuture<IBasicPluginInformation[]> {
+		return (() => {
+			let nativescriptUrl = `http://npmsearch.com/query?fields=name,version,description&sort=rating+desc&q=name:${name}&start=0&size=100`;
+			let result = this.$httpClient.httpRequest(nativescriptUrl).wait().body;
+			if(result) {
+				let npmSearchResult: any[] = JSON.parse(result).results;
+				let plugins =_.map(npmSearchResult, pluginResult => {
+					let pluginInfo: IBasicPluginInformation = {
+						name: this.getStringFromNpmSearchResult(pluginResult, "name"),
+						version: this.getStringFromNpmSearchResult(pluginResult, "version"),
+						description: this.getStringFromNpmSearchResult(pluginResult, "description")
+					};
+					return pluginInfo;
+				});
+
+				return plugins;
+			}
+			return null;
+		}).future<IBasicPluginInformation[]>()();
+	}
+}
+$injector.register("nativeScriptProjectPluginsService", NativeScriptProjectPluginsService);
+
+export enum PluginType {
+	NpmPlugin = 0,
+	NpmNativeScriptPlugins = 1,
+	MarketplacePlugin = 2
+}
+
+export class NativeScriptPluginData implements IPlugin {
+	public configurations: string[];
+
+	constructor(public data: IPluginInfoBase,
+		public type: PluginType,
+		protected $project: Project.IProject) {
+		this.configurations = [];
+	}
+
+	public get pluginInformation(): string[] {
+		let additionalPluginData: string[];
+		if(this.data.Platforms && this.data.Platforms.length > 0) {
+			additionalPluginData = [this.buildRow("Platforms", this.data.Platforms.join(", "))];
+		}
+		return this.composePluginInformation(additionalPluginData);
+	}
+
+	public toProjectDataRecord(version: string): string {
+		return this.data.Name;
```

Comment:
is this correct? The plugin will not be added in the project (`.abproject`) at all